tracer/test/Datadog.Trace.Tools.Runner.IntegrationTests/BaseRunCommandTests.cs

```diff
@@ -0,0 +1,122 @@
+// <copyright file="BaseRunCommandTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Datadog.Trace.Tools.Runner.IntegrationTests
+{
+    public abstract class BaseRunCommandTests
+    {
+        protected BaseRunCommandTests(string commandPrefix, bool enableCiVisibilityMode)
+        {
+            CommandPrefix = commandPrefix;
+            EnableCiVisibilityMode = enableCiVisibilityMode;
+        }
+
+        protected string CommandPrefix { get; }
+
+        protected bool EnableCiVisibilityMode { get; }
+
+        [Fact]
+        public void Run()
+        {
+            string command = null;
+            string arguments = null;
+            Dictionary<string, string> environmentVariables = null;
+            bool callbackInvoked = false;
+
+            Program.CallbackForTests = (c, a, e) =>
+            {
+                command = c;
+                arguments = a;
+                environmentVariables = e;
+                callbackInvoked = true;
+            };
+
+            // CI visibility mode checks if there's a running agent
+            using var agent = EnableCiVisibilityMode ? new MockTracerAgent(TcpPortProvider.GetOpenPort()) : null;
+
+            var agentUrl = $"http://localhost:{agent?.Port ?? 1111}";
+
+            var commandLine = $"{CommandPrefix} test.exe --dd-env TestEnv --dd-service TestService --dd-version TestVersion --tracer-home TestTracerHome --agent-url {agentUrl} --set-env VAR1=A --set-env VAR2=B";
+
+            using var console = ConsoleHelper.Redirect();
+
+            var exitCode = Program.Main(commandLine.Split(' '));
+
+            using var scope = new AssertionScope();
+
+            scope.AddReportable("output", console.Output);
+
+            exitCode.Should().Be(0);
+            callbackInvoked.Should().BeTrue();
+
+            command.Should().Be("test.exe");
+            arguments.Should().BeNullOrEmpty();
+            environmentVariables.Should().NotBeNull();
+
+            environmentVariables["DD_ENV"].Should().Be("TestEnv");
+            environmentVariables["DD_SERVICE"].Should().Be("TestService");
+            environmentVariables["DD_VERSION"].Should().Be("TestVersion");
+            environmentVariables["DD_DOTNET_TRACER_HOME"].Should().Be("TestTracerHome");
+            environmentVariables["DD_TRACE_AGENT_URL"].Should().Be(agentUrl);
+            environmentVariables["VAR1"].Should().Be("A");
+            environmentVariables["VAR2"].Should().Be("B");
+
```

Comment:
These will throw if they don't exist, so prob better to use fluent assertions directly
```suggestion
            environmentVariables.Should().Contain("DD_ENV", "TestEnv");
            environmentVariables.Should().Contain("DD_SERVICE", "TestService");
            environmentVariables.Should().Contain("DD_VERSION", "TestVersion");
            environmentVariables.Should().Contain("DD_DOTNET_TRACER_HOME", "TestTracerHome");
            environmentVariables.Should().Contain("DD_TRACE_AGENT_URL", agentUrl);
            environmentVariables.Should().Contain("VAR1", "A");
            environmentVariables.Should().Contain("VAR2", "B");
```

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2295/commits/54c8e313695aae194ab141e73e2b033a62f3f06b